### PR TITLE
Updating to okta-vue v5

### PR DESCRIPTION
**Pull request by bdemers**
Updating sample code from upstream project (lots of minor formatting changes)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,23 +9,23 @@
...
   "devDependencies": {
     "@vue/cli-plugin-babel": "~4.5.13",
     "@vue/cli-plugin-eslint": "~4.5.13",
     "@vue/cli-plugin-router": "~4.5.13",
     "@vue/cli-service": "~4.5.13",
-    "@vue/compiler-sfc": "^3.1.1",
+    "@vue/compiler-sfc": "^3.1.2",
     "babel-eslint": "^10.1.0",
-    "eslint": "^7.32.0",
+    "eslint": "^6.7.2",
```
Comment (bdemers) [inline package.json:28]: @mraible eslint v7 didn't work for me, is this expected?

Comment (mraible) [reply package.json:28]: It works for me:

```
$ npm run lint

> okta-vue-sample@1.0.0 lint /Users/mraible/dev/okta-samples/okta-vue-sample
> vue-cli-service lint

 DONE  No lint errors found!
```

What fails for you?

File: src/App.vue [hunk 2]
```diff
@@ -15,56 +15,56 @@
...
           >
           &nbsp;
           Okta-Vue Sample Project
         </router-link>
         <a
-          class="item"
-          v-if="!authState.isAuthenticated"
-          v-on:click="login()"
+            class="item"
+            v-if="authState && !authState.isAuthenticated"
```
Comment (mraible) [inline src/App.vue:30]: You should be able to shorten this to `!authState?.isAuthenticated`

File: src/App.vue [hunk 3]
```diff
@@ -15,56 +15,56 @@
...
         </a>
         <router-link
-          to="/messages"
-          class="item"
-          id="messages-button"
-          v-if="authState.isAuthenticated"
+            to="/messages"
+            class="item"
+            id="messages-button"
+            v-if="authState && authState.isAuthenticated"
```
Comment (mraible) [inline src/App.vue:39]:
```suggestion
            v-if="authState?.isAuthenticated"
```

File: src/App.vue [hunk 4]
```diff
@@ -15,56 +15,56 @@
...
         </router-link>
         <router-link
-          to="/profile"
-          class="item"
-          id="profile-button"
-          v-if="authState.isAuthenticated"
+            to="/profile"
+            class="item"
+            id="profile-button"
+            v-if="authState && authState.isAuthenticated"
```
Comment (mraible) [inline src/App.vue:51]:
```suggestion
            v-if="authState?.isAuthenticated"
```

File: src/App.vue [hunk 5]
```diff
@@ -15,56 +15,56 @@
...
+          Profile
         </router-link>
         <a
-          id="logout-button"
-          class="item"
-          v-if="authState.isAuthenticated"
-          v-on:click="logout()"
+            id="logout-button"
+            class="item"
+            v-if="authState && authState.isAuthenticated"
```
Comment (mraible) [inline src/App.vue:58]:
```suggestion
            v-if="authState?.isAuthenticated"
```

File: src/components/Home.vue [hunk 6]
```diff
@@ -13,7 +13,7 @@
 <template>
   <div id="home">
     <h1 class="ui header">PKCE Flow w/ Okta Hosted Login Page</h1>
-    <div v-if="!authState.isAuthenticated">
+    <div v-if="!authState || !authState.isAuthenticated">
```
Comment (mraible) [inline src/components/Home.vue:16]:
```suggestion
    <div v-if="!authState?.isAuthenticated">
```

File: src/components/Home.vue [hunk 7]
```diff
@@ -22,16 +22,16 @@
...
+          class="ui primary button"
+          role="button"
+          v-on:click="login()"
       >
         Login
       </button>
     </div>
 
-    <div v-if="authState.isAuthenticated">
+    <div v-if="authState && authState.isAuthenticated">
```
Comment (mraible) [inline src/components/Home.vue:34]:
```suggestion
    <div v-if="authState?.isAuthenticated">
```

File: src/components/Home.vue [hunk 8]
```diff
@@ -84,12 +84,12 @@ export default {
   created () { this.setup() },
   methods: {
     async setup () {
-      if (this.authState.isAuthenticated) {
+      if (this.authState && this.authState.isAuthenticated) {
```
Comment (mraible) [inline src/components/Home.vue:87]:
```suggestion
      if (this.authState?.isAuthenticated) {
```

**Review by mraible — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (bdemers) [reply package.json:28]: ugg, sorry, my original message was literally the classic terrible "it didn't work"

It was a dependency resolution/compatibility issue when running `npm install`

File: src/App.vue (hunk 2 above)
Comment (bdemers) [inline src/App.vue:30]:
```suggestion
            v-if="!authState.isAuthenticated"
```

Comment (mraible) [reply src/App.vue]: Confirmed this works in OktaDev Schematics just a few minutes ago. https://github.com/oktadev/schematics/pull/638

File: package.json (hunk 1 above)
Comment (bdemers) [reply package.json:28]: @mraible 

Looks like vue is stuck at eslint v6 until the next version of vue v5?

https://github.com/vuejs/vue-cli/blob/v4/packages/%40vue/cli-plugin-eslint/package.json#L35

```txt
npm install

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: eslint@7.32.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"^7.32.0" from the root project
npm ERR!   peer eslint@">= 4.12.1" from babel-eslint@10.1.0
npm ERR!   node_modules/babel-eslint
npm ERR!     dev babel-eslint@"^10.1.0" from the root project
npm ERR!   2 more (eslint-plugin-vue, vue-eslint-parser)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@">= 1.6.0 < 7.0.0" from @vue/cli-plugin-eslint@4.5.13
npm ERR! node_modules/@vue/cli-plugin-eslint
npm ERR!   dev @vue/cli-plugin-eslint@"~4.5.13" from the root project
```

My dependency fu isn't as strong in the JS world, any ideas? (all i did to repro was to revert the eslint version)

Comment (mraible) [reply package.json:28]: It looks like you're using npm v7, that's why you see this issue. It doesn't happen for me with npm 6. This is likely because npm 6 didn't enforce peer dependencies like npm v7 does. You can workaround this by adding an `.npmrc` to the project and putting the following in it:

```
legacy-peer-deps = true
```

Of course, not upgrading to ESLint 7 works too!

**Review by mraible — APPROVED** [review]
